code/Gamemodes/Modes/FFA/FreeForAll.cs

```diff
@@ -156,34 +159,38 @@ private async ValueTask<bool> CleanupTurn()
 
 	private async Task HandleGrubDeaths()
 	{
-		bool rerun;
-		do
+		foreach ( var player in Players )
 		{
-			rerun = false;
-			foreach ( var player in PlayerList )
+			if ( player.IsDead )
+				continue;
+
+			foreach ( var grub in player.Grubs )
 			{
-				if ( player.Dead )
+				if ( grub.LifeState == LifeState.Dead )
 					continue;
 
-				foreach ( var grub in player.Grubs )
-				{
-					if ( grub.LifeState == LifeState.Dead )
-						continue;
+				if ( player.IsDisconnected )
+					grub.TakeDamage( DamageInfo.Generic( float.MaxValue ).WithTag( "disconnect" ) );
 
-					if ( !grub.HasBeenDamaged )
-						continue;
-
-					rerun = true;
-					if ( grub.ApplyDamage() && grub.DeathTask is not null && !grub.DeathTask.IsCompleted )
-						await grub.DeathTask;
+				if ( !grub.HasBeenDamaged )
+					continue;
 
-					CameraTarget = grub;
-					// TODO: Send an event to the UI for grub damage worldhud.
-					await GameTask.Delay( 1000 );
-					CameraTarget = null;
-				}
+				await HandleGrubDeath( grub );
 			}
-		} while ( rerun );
+		}
+	}
+
+	private async Task HandleGrubDeath( Grub grub )

```

Comment:
Different function name for this? The previous code felt like it had too much code in the function.

Reply:
HandleGrubDeath is fine. It's just a logical continuation of the previous function.